os/board/common/partitions.c

```diff
@@ -114,26 +115,31 @@ static int type_specific_initialize(int minor, FAR struct mtd_dev_s *mtd_part, i
 
 #ifdef CONFIG_MTD_FTL
 	bool do_ftlinit = false;
-	if (!strncmp(types, "ftl,", 4)
+	if (!strncmp(types, "ftl,", 4)) {
+		tagno = MTD_FTL;
+		do_ftlinit = true;
+	}
 #ifdef CONFIG_BINARY_MANAGER
-	|| !strncmp(types, "kernel,", 7)
+	else if (!strncmp(types, "kernel,", 7)
 	|| !strncmp(types, "bootparam,", 10)
 #ifdef CONFIG_APP_BINARY_SEPARATION
```

Comment:
If enabled `CONFIG_BINARY_MANAGER`, but not enabled `CONFIG_APP_BINARY_SEPARATION`, it makes build error.
Please check this.